src/fuzzy_cmeans.jl

```diff
@@ -0,0 +1,61 @@
+function update_weights!(weights, data, centers, fuzziness, norm_fn)
+  pow = 2.0/(fuzziness-1)
+  ly, lx = length.(indices(weights))
```

Comment:
Better as
```julia
indcols, indrows = indices(weights)
for i in indrows
...
```
(It's a bit inconsistent to use `indices` to support general indices and then use `1:ly` for indexing.)

Even better might be to use some informative names for these variables, e.g., `indpoints` and `indclusters`.

This issue is applicable throughout.